openml-h2o/src/test/java/com/feedzai/openml/h2o/H2OModelParamValidationTest.java

```diff
@@ -0,0 +1,124 @@
+ *
+ * @since 1.2.1
+ */
+@RunWith(Parameterized.class)
+public class H2OModelParamValidationTest {
+
+    private final H2OAlgorithm h2OAlgorithm;
+    private final Map<String, String> defaultAlgoParams;
+    private final String badParamName;
+    private final String badParamValue;
+
+    public H2OModelParamValidationTest(final H2OAlgorithm h2OAlgorithm,
+                                       final Map<String, String> defaultAlgoParams,
+                                       final String badParamName,
+                                       final String badParamValue) {
+        this.h2OAlgorithm = h2OAlgorithm;
+        this.defaultAlgoParams = defaultAlgoParams;
+        this.badParamName = badParamName;
+        this.badParamValue = badParamValue;
+    }
+
+    @Parameterized.Parameters
+    public static Collection<Object[]> data() {
+        return Arrays.asList(new Object[][] {
+                // { algorithm, default parameters, bad parameter name, bad parameter value }
+                { H2OAlgorithm.NAIVE_BAYES_CLASSIFIER, H2OAlgorithmTestParams.getBayes(), "laplace", "-0.25" },
+                { H2OAlgorithm.ISOLATION_FOREST, H2OAlgorithmTestParams.getIsolationForest(), "mtries", "-3" },
+                { H2OAlgorithm.XG_BOOST, H2OAlgorithmTestParams.getXgboost(), "learn_rate", "2" },
+                { H2OAlgorithm.DEEP_LEARNING, H2OAlgorithmTestParams.getDeepLearning(), "mini_batch_size", "0" },
+                { H2OAlgorithm.DISTRIBUTED_RANDOM_FOREST, H2OAlgorithmTestParams.getDrf(), "mtries", "-3" },
+                { H2OAlgorithm.GRADIENT_BOOSTING_MACHINE, H2OAlgorithmTestParams.getGbm(), "max_abs_leafnode_pred", "0" },
+                { H2OAlgorithm.GENERALIZED_LINEAR_MODEL, H2OAlgorithmTestParams.getGlm(), "alpha", "2" }
+        });
+    }
+
+
+    /**
+     * Tests H20 parameter validation when all parameters correct.
+     */
+    @Test
+    public final void testValidationOKWhenAllParametersAreValid() {
+        final List<ParamValidationError> validationErrors = validateParamsForAlgorithm(defaultAlgoParams, h2OAlgorithm);
+
+        assertThat(validationErrors).isEmpty();
+    }
+
+    /**
+     * Tests H20 parameter validation for with one parameter invalid.
+     */
+    @Test
+    public final void testHasValidationErrorsWhenParametersAreInvalid() {
+        final Map<String, String> params = new HashMap<>(defaultAlgoParams);
+        params.put(badParamName, badParamValue);
+
+        final List<ParamValidationError> validationErrors = validateParamsForAlgorithm(params, h2OAlgorithm);
+
+        assertThat(validationErrors)
+                .isNotEmpty()
+                .allMatch(error -> error.getMessage().contains("Model has errors: "));
+    }
+
+    /**
+     * Validates params for a given algorithm
+     * @param params     The parameters to validate
+     * @param algorithm  The H2O algorithm
+     * @return           List of param validation errors
+     *
```

Comment:
the description of return tag doesn't need to be aligned with the description of the parameters, and the javadoc/descriptions (both from parameters and returns) should end with a final mark.

e.g.
`@return List of param validation errors`

also remove the last empty line of the javadoc

Reply:
Addressed all Java-doc issues, thanks for the heads up!